components/org.wso2.carbon.identity.keyrotation/src/main/java/org.wso2.carbon.identity.keyrotation/config/KeyRotationConfig.java

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.keyrotation.config;
+
+import org.apache.axiom.om.util.Base64;
+import org.apache.log4j.Logger;
+import org.wso2.carbon.identity.keyrotation.util.KeyRotationException;
+import org.wso2.carbon.identity.keyrotation.util.KeyRotationServiceUtils;
+
+import java.io.File;
+import java.lang.reflect.Field;
+import java.nio.file.Paths;
+
+/**
+ * This class holds implementations needed to load the configurations in the properties.yaml file.
+ */
+public class KeyRotationConfig {
+
+    private String oldSecretKey;
+    private String newSecretKey;
+    private String newISHome;
+    private String oldIdnDBUrl;
+    private String oldIdnUsername;
+    private String oldIdnPassword;
+    private String newIdnDBUrl;
+    private String newIdnUsername;
+    private String newIdnPassword;
+    private String newRegDBUrl;
+    private String newRegUsername;
+    private String newRegPassword;
+    private String enableDBMigrator;
+    private String enableConfigMigrator;
+    private String enableSyncMigrator;
+    private static final Logger log = Logger.getLogger(KeyRotationConfig.class);
+    private static final KeyRotationConfig instance = new KeyRotationConfig();
+
+    public static KeyRotationConfig getInstance() {
+
+        return instance;
+    }
+
+    /**
+     * Load the configurations placed in the properties.yaml file.
+     *
+     * @return KeyRotation config object.
+     * @throws KeyRotationException Exception thrown while loading the configs from properties.yaml file.
+     */
+    public KeyRotationConfig loadConfigs(String[] args) throws KeyRotationException {
+
+        String propertiesFilePath = Paths.get("components", "org.wso2.carbon.identity.keyrotation", "src",
```

Comment:
If properties.yaml file does not exist what will happen?

Reply:
It will take the default file values inside the resources directory